File: 0169-majority-element/0169-majority-element.java
import java.util.Map.Entry;

class Solution {
    public int majorityElement(int[] nums) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int num : nums) {
          map.put(num, map.getOrDefault(num, 0) + 1);
        }

        Entry<Integer, Integer> first = map.entrySet().stream()
            .filter(o -> o.getValue() > nums.length / 2).findFirst().get();

        return first.getKey();
    }
}

File: 0169-majority-element/0169-majority-element.py
from collections import Counter

class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        # solved O(1)
        # 같은 값 -> count++, 다른 값-> count--
        # if count == 0 -> re-set major
        response, count = 0, 0
        for num in nums:
            # initially set
            if count == 0:
                response = num
            count += 1 if num == response else -1
        return response